oak_remote_attestation_verification/src/verifier.rs

```diff
@@ -77,59 +72,81 @@ impl AttestationVerifier for InsecureAttestationVerifier {
     }
 }
 
-/// Verifies the transparent release endorsement for a given measurement, using the given
-/// verification options to control the extent of the verification. Summary of the verification
-/// logic:
+const PEM_HEADER: &str = "-----BEGIN PUBLIC KEY-----";
+const PEM_FOOTER: &str = "-----END PUBLIC KEY-----";
+
+/// Makes a plausible guess whether the public key is in PEM format.
+pub fn looks_like_pem(maybe_pem: &str) -> bool {
+    let p = maybe_pem.trim();
+    p.starts_with(PEM_HEADER) && p.ends_with(PEM_FOOTER)
+}
+
+/// Converts a pem key to raw. Will panic if it does not like like pem.
+pub fn convert_pem_to_raw(public_key_pem: &str) -> anyhow::Result<Vec<u8>> {
+    let stripped = public_key_pem
+        .trim()
+        .strip_prefix(PEM_HEADER)
+        .expect("could not find expected header")
+        .strip_suffix(PEM_FOOTER)
+        .expect("could not find expected footer");
+    let remove_newlines = stripped.replace('\n', "");
+
+    Ok(BASE64_STANDARD.decode(remove_newlines)?)
+}
+
+pub fn convert_raw_to_pem(public_key: &[u8]) -> String {
```

Comment:
I think conversion logic could be moved to a separate file. And `verifier.rs` will only contain the verifier logic.

Reply:
I agree. Let's put this off until later to reduce back and forth. Also, I couldn't wrap my head around the split between rekor.rs and verifier.rs. These could be unified into verifier.rs.

Reply:
I think Transparent Release related code could be moved to a `mod transparent_release_verification`, and the `verifier.rs` will have a `Verifier` class with a `fn verify(Evidence, Endorsement, ReferenceValue)` method that implements a full DICE attestation verification

Reply:
Actually I was thinking to have fewer mods not more. IIUC criteria whether to join (or divide) two features as Rust modules: (1) differences in dependencies. If one feature pulls in a dep and the other not, then this indicates that there could be two modules. If they have the same deps, then they can be in the same module. (2) Users are picky about size, and one feature blows up the size too much for users of the other feature. But I think the linker would eliminate unused symbols anyway, so it is more an optimization of the compile process. (3) Topics of the features is too unrelated, so they simply don't belong together.

According to (1), (2), (3) we could have even a single (and more beefy) module "attestation" which contains the verification as well. It would have maybe 10 *.rs files in it. What would be the downsides of this, based on Rust language and Oak needs?

Reply:
We won't need `attestation` on the client side, so I think it's fine to keep 2 separate attestation and verification crates